### DISCUSSION TITLE
How to run fastapi with uvicorn in digitalocean app platform

### DISCUSSION THREAD

**Discussion by ghost** [Kludex/uvicorn#2029] [Q&A] [answered]
Hello development team,
I am unable to run my FastAPI app on the DigitalOcean App Platform. Although my app starts successfully, it fails during the deployment phase.
![photo_2023-07-05_20-32-38](https://github.com/encode/uvicorn/assets/30009591/34f513f5-fd08-4727-aff6-ccc462dbc789)

my digitalocean run command is:

```
python main.py
```

my main.py file code is:

```
from fastapi import FastAPI
from starlette.middleware.cors import CORSMiddleware
from starlette.middleware.trustedhost import TrustedHostMiddleware
from endpoints.subscribe import router
import uvicorn

app = FastAPI()
app.include_router(router)
CORS_ALLOWED_ORIGINS = ["test.com", "www.test.com"]


@app.get("/")
async def root():
    return {"status": "success", "message": "Ok"}


app.add_middleware(
    CORSMiddleware,
    allow_origins=CORS_ALLOWED_ORIGINS,
    allow_credentials=True,
    allow_methods=["*"],
    allow_headers=["*"],
)
app.add_middleware(TrustedHostMiddleware, allowed_hosts=["*"])

if __name__ == "__main__":
    uvicorn.run(app, host='127.0.0.1', port=8000)
```

**Comment by Kludex** [comment, marked as answer]
I don't know... But uvicorn is working.

Ask Digital Ocean: https://docs.digitalocean.com/support/

**Reply by ghost** [reply]
I think that uvicorn is unable to expose the outbound port.

**Comment by ghost** [comment]
i fixed it. you can use this command:

```
uvicorn main:app --host=0.0.0.0 --port=${PORT:-8000}
```

**Reply by Zirochkaa** [reply]
Thanks, @amirshnll, I had the same problem and your solution helped 👍

**Comment by ghost** [comment]
> Thanks, @amirshnll, I had the same problem and your solution helped 👍

🚀
